### PR TITLE
fix(metadata-sidebar): Fix metadata sidebar ai capabilities

### PR DESCRIPTION
**Pull request by jankowiakdawid**
Fix from where we read if Metadata Sidebar should have AI capabilities turned on.

### Screenshots

Screenshot 2024-09-26 at 15 01 16

Screenshot 2024-09-26 at 15 01 53

Screenshot 2024-09-26 at 15 02 50

Screenshot 2024-09-26 at 15 20 36

### CODE REVIEW

**Review by wpiesiak — DISMISSED** [review]

File: src/elements/content-sidebar/MetadataSidebarRedesign.tsx [hunk 1]
```diff
@@ -64,14 +64,7 @@ export interface MetadataSidebarRedesignProps extends PropsWithoutContext, Error
...
 
-function MetadataSidebarRedesign({
-    api,
-    elementId,
-    fileId,
-    isBoxAiSuggestionsEnabled,
-    onError,
-    isFeatureEnabled,
-}: MetadataSidebarRedesignProps) {
+function MetadataSidebarRedesign({ api, elementId, fileId, onError, isFeatureEnabled }: MetadataSidebarRedesignProps) {
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/MetadataSidebarRedesign.tsx:67]: is `isFeatureEnabled` doing what `useFeatureEnabled` is doing? do we need both?

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 2]
```diff
@@ -139,6 +138,10 @@ export const EmptyStateWithBoxAiEnabled: StoryObj<typeof MetadataSidebarRedesign
         metadataSidebarProps: {
             ...defaultMetadataSidebarProps,
         },
+        features: {
+            'metadata.redesign.enabled': true,
+            'metadata.aiSuggestions.enabled': true,
+        },
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:144]:
```suggestion
        features: {
           ...mockFeatures,
            'metadata.aiSuggestions.enabled': true,
        },
```

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 3]
```diff
@@ -214,3 +220,28 @@ export const MetadataInstanceEditorCancelChanges: StoryObj<typeof MetadataSideba
         );
     },
 };
+
+export const MetadataInstanceEditorAIEnabled: StoryObj<typeof MetadataSidebarRedesign> = {
+    args: {
+        features: {
+            'metadata.redesign.enabled': true,
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:227]:
```suggestion
            ...mockFeatures,
```

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

File: src/elements/content-sidebar/MetadataSidebarRedesign.tsx (hunk 1 above)
Comment (jankowiakdawid) [reply src/elements/content-sidebar/MetadataSidebarRedesign.tsx:67]: No I don't think so.

`useFeatureEnabled` is checking the `features` array. In our case we can check if metadata redesign is enabled, or if metadata aiSuggestions are enabled.

`isFeatureEnabled` is a prop that comes from host application in `metadataSidebarProps`
https://github.com/box/box-ui-elements/blob/549188a9f005b8dd0f63c0ed4906e8a64144c3a4/src/elements/content-sidebar/SidebarPanels.js#L61

it's defined here https://github.com/box/box-ui-elements/blob/549188a9f005b8dd0f63c0ed4906e8a64144c3a4/src/elements/content-sidebar/MetadataSidebar.js#L43-L47

maybe @bfoxx1906 or @jstoffan have more insight if it should be used 🤔

Comment (JChan106) [reply src/elements/content-sidebar/MetadataSidebarRedesign.tsx:67]: Took a look, useFeatureEnabled isn't used anywhere else  in BUIE. However, Seems like a use case for components that don't have the features array passed down, so it probably is needed here. Have we tested useFeatureEnabled works as expected here?

Comment (jankowiakdawid) [reply src/elements/content-sidebar/MetadataSidebarRedesign.tsx:67]: Yes, `useFeatureEnabled` works as expected. It's covered with a storybook story test (also in this PR). It was recommended by @jstoffan some time ago.

Now after re-reading this I understand initial confusion. As I mentioned `isFeatereEnalbed` is a prop indicating if user has __Metadata Sidebar__ enabled for them.

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 4]
```diff
@@ -188,14 +191,17 @@ export const MetadataInstanceEditorCancelChanges: StoryObj<typeof MetadataSideba
     play: async ({ canvasElement }) => {
         const canvas = within(canvasElement);
 
-        const editButtons = await canvas.findAllByRole('button', { name: 'Edit' }, { timeout: 5000 });
+        // Edit buttons contains also template name
+        const editButtons = await canvas.findAllByRole('button', { name: /^edit/i }, { timeout: 5000 });
+        expect(editButtons).toHaveLength(3);
```
Comment (TylerGauntlett) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:196]: ❓ any reason to using `findAllByRole` here? I see the length check below but most everything else only depends on the first button.  Should the selector be updated to using `Edit My Template` directly?

Comment (jankowiakdawid) [reply src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:196]: It definitely can, I didn't wanted to change too much in the test. Just fix it.

It started failing after we did fixes for A11y in `metadata-editor` and the `aria-label` for the Edit button got a template name.

File: src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx [hunk 5]
```diff
@@ -188,16 +191,19 @@ export const MetadataInstanceEditorCancelChanges: StoryObj<typeof MetadataSideba
...
+        const editButton = await canvas.findByRole('button', { name: 'Edit My Template' }, { timeout: 5000 });
+        expect(editButton).toBeInTheDocument();
 
         let headlines = await canvas.findAllByRole('heading', { level: 1 });
         expect(headlines).toHaveLength(3);
         expect(headlines.map(heading => heading.textContent)).toEqual(
             expect.arrayContaining(['My Template', 'Select Dropdowns', 'Custom Metadata']),
         );
 
+        expect(editButton.ariaLabel).toBe('Edit My Template');
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/stories/tests/MetadataSidebarRedesign-visual.stories.tsx:204]: I dont think we need to assert for it again, `findByRole` is using that label to locate that `editButton`

File: src/elements/content-sidebar/__tests__/MetadataSidebarRedesign.test.tsx [hunk 6]
```diff
@@ -54,17 +55,20 @@ describe('elements/content-sidebar/Metadata/MetadataSidebarRedesign', () => {
...
             fileId: 'test-file-id-1',
             elementId: 'element-1',
-            isBoxAiSuggestionsEnabled: true,
             isFeatureEnabled: true,
             onError: jest.fn(),
         } satisfies MetadataSidebarRedesignProps;
 
-        render(<MetadataSidebarRedesign {...defaultProps} {...props} />);
+        render(
+            <FeatureProvider features={features}>
```
Comment (greg-in-a-box) [inline src/elements/content-sidebar/__tests__/MetadataSidebarRedesign.test.tsx:68]: I think we might able to move this provider into the testing-utils, and still be able to pass in the features object similar how the children are being passed in

**Review by greg-in-a-box — CHANGES_REQUESTED** [review]

**Review by greg-in-a-box — DISMISSED** [review]

**Review by TylerGauntlett — APPROVED** [review]

**Review by greg-in-a-box — APPROVED** [review]

**Pull request merged**